### ISSUE TITLE
is this "thread safe"?

### ISSUE THREAD

**Issue by thinkong** [manucorporat/stats#1]
Hi.. I know this was.. maybe not meant to be used.. but I saw this from the gin examples and thought maybe i would use it?

But then.. is this threadsafe? I noticed that there are no mutex/locks here or anywhere in the examples in gin

**Comment by manucorporat** [comment]
@thinkong yes, it uses a reader/writer mutual exclusion lock, it is thread safe.
https://github.com/manucorporat/stats/blob/master/stats.go#L9
